ui/components/app/wallet-overview/aggregated-percentage-overview.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { zeroAddress } from 'ethereumjs-util';
+import {
+  getCurrentCurrency,
+  getSelectedAccount,
+  getShouldHideZeroBalanceTokens,
+  getTokensMarketData,
+} from '../../../selectors';
+
+import { useAccountTotalFiatBalance } from '../../../hooks/useAccountTotalFiatBalance';
+import { isEqualCaseInsensitive } from '../../../../shared/modules/string-utils';
+import { formatValue, isValidAmount } from '../../../../app/scripts/lib/util';
+import { getIntlLocale } from '../../../ducks/locale/locale';
+import {
+  Display,
+  TextColor,
+  TextVariant,
+  FontWeight,
+} from '../../../helpers/constants/design-system';
+import { Box, Text } from '../../component-library';
+import { getCalculatedTokenAmount1dAgo } from '../../../helpers/utils/util';
+
+export const AggregatedPercentageOverview = () => {
+  const tokensMarketData = useSelector(getTokensMarketData);
+  const locale = useSelector(getIntlLocale);
+  const fiatCurrency = useSelector(getCurrentCurrency);
+  const selectedAccount = useSelector(getSelectedAccount);
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+  // Get total balance (native + tokens)
+  const { totalFiatBalance, orderedTokenList } = useAccountTotalFiatBalance(
+    selectedAccount,
+    shouldHideZeroBalanceTokens,
+  );
+
+  let totalFiat1dAgo = 0;
+  orderedTokenList.forEach((token) => {
```

Comment:
This section of the code, which contains two loops, can be resource-intensive. It would be preferable to wrap it in a useMemo hook to ensure that the calculation is only performed when one of its dependencies changes, rather than on every render.

Reply:
it's can be extracted to a separated hook as well , then use memorization inside this hook

Reply:
Memoize done in [81defac](https://github.com/MetaMask/metamask-extension/pull/27108/commits/81defaca2e6258d4beb9d85c255f2d7c84096ec1)